python/lsst/pipe/tasks/healSparseMappingProperties.py

```diff
@@ -0,0 +1,490 @@
+
+        Parameters
+        ----------
+        total_weights : `np.ndarray`
+            Total accumulated weights, for each value index.
+        total_inputs : `np.ndarray`
+            Total number of inputs, for each value index.
+        """
+        if self.config.do_mean:
+            use, = np.where(total_inputs > 0)
+            self.mean_values[use] /= total_inputs[use]
+        if self.config.do_weighted_mean:
+            use, = np.where(total_weights > 0.0)
+            self.weighted_mean_values[use] /= total_weights[use]
+
+        # And perform any necessary post-processing
+        self.post_process(total_weights, total_inputs)
+
+    def set_map_values(self, pixels):
+        """Assign accumulated values to the maps.
+
+        Parameters
+        ----------
+        pixels : `np.ndarray`
+            Array of healpix pixels (nest scheme) to set in the map.
+        """
+        if self.config.do_min:
+            self.min_map[pixels] = self.min_values
+        if self.config.do_max:
+            self.max_map[pixels] = self.max_values
+        if self.config.do_mean:
+            self.mean_map[pixels] = self.mean_values
+        if self.config.do_weighted_mean:
+            self.weighted_mean_map[pixels] = self.weighted_mean_values
+        if self.config.do_sum:
+            self.sum_map[pixels] = self.sum_values
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        """Compute map value from a row in the visitSummary catalog.
+
+        Parameters
+        ---------
+        row : `lsst.afw.table.ExposureRecord`
+            Row of a visitSummary ExposureCatalog.
+        ra : `np.ndarray`
+            Array of right ascensions
+        dec : `np.ndarray`
+            Array of declinations
+        scalings : `float` or `np.ndarray`
+            Scaling values to coadd zeropoint.
+        psf_array : `np.ndarray`, optional
+            Array of approximate psf values matched to ra/dec.
+        """
+        raise NotImplementedError("All property maps must implement compute()")
+
+    def post_process(self, total_weights, total_inputs):
+        """Perform post-processing on values.
+
+        Parameters
+                ----------
+        total_weights : `np.ndarray`
+            Total accumulated weights, for each value index.
+        total_inputs : `np.ndarray`
+            Total number of inputs, for each value index.
+        """
+        # Override of this method is not required.
+        pass
+
+
+@register_property_map("exposure_time")
+class ExposureTimePropertyMap(BasePropertyMap):
+    """Exposure time property map."""
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return row.getVisitInfo().getExposureTime()
+
+
+@register_property_map("psf_size")
+class PsfSizePropertyMap(BasePropertyMap):
+    """PSF size property map."""
+    requires_psf = True
+
+    def compute(self, row, ra, dec, scalings, psf_array=None):
+        return psf_array["psf_size"]
```

Comment:
This will crash if `psf_array` is `None`, which would be bad, but that is a programming logic error not a user error.  I've added some checks, and I think that if `compute` is instead `_compute` then it'll be clearer that this isn't something that a user should be interacting with anyway.